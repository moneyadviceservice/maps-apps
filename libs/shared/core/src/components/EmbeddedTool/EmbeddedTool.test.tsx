import React from 'react';

import { render, screen } from '@testing-library/react';

import { EmbeddedTool } from '.';

import '@testing-library/jest-dom/extend-expect';

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({
    query: {},
  }),
}));

describe('Embedded Tool component', () => {
  it('renders correctly', () => {
    render(
      <EmbeddedTool
        toolData={{
          url: {
            en: 'https://dev-moneyhelper-tools.azurewebsites.net/en/leave-pot-untouched',
            cy: 'https://dev-moneyhelper-tools.azurewebsites.net/cy/leave-pot-untouched',
          },
          api: 'https://dev-moneyhelper-tools.azurewebsites.net/api/embed',
          id: 'leave-pot-untouched',
        }}
        testId="test-component"
      />,
    );
    const container = screen.getByTestId('test-component');
    expect(container).toMatchSnapshot();
  });

  it('renders correctly with default id', () => {
    render(
      <EmbeddedTool
        toolData={{
          url: {
            en: 'https://dev-moneyhelper-tools.azurewebsites.net/en/leave-pot-untouched',
            cy: 'https://dev-moneyhelper-tools.azurewebsites.net/cy/leave-pot-untouched',
          },
          api: 'https://dev-moneyhelper-tools.azurewebsites.net/api/embed',
          id: 'leave-pot-untouched',
        }}
      />,
    );
    const container = screen.getByTestId('iframe');
    expect(container).toMatchSnapshot();
  });
});
