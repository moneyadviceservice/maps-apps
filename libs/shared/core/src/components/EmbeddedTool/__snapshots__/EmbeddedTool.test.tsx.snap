// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Embedded Tool component renders correctly 1`] = `
<iframe
  class="mas-iframe"
  data-testid="test-component"
  id="leave-pot-untouched"
  src="https://dev-moneyhelper-tools.azurewebsites.net/en/leave-pot-untouched"
  style="width: 100%; height: 100%; left: 0px; position: absolute; top: 0px; border: 0px;"
  title="leave-pot-untouched"
/>
`;

exports[`Embedded Tool component renders correctly with default id 1`] = `
<iframe
  class="mas-iframe"
  data-testid="iframe"
  id="leave-pot-untouched"
  src="https://dev-moneyhelper-tools.azurewebsites.net/en/leave-pot-untouched"
  style="width: 100%; height: 100%; left: 0px; position: absolute; top: 0px; border: 0px;"
  title="leave-pot-untouched"
/>
`;
